iot-hub/Samples/device/X509DeviceCertWithChainSample/readme.md

```diff
@@ -0,0 +1,21 @@
+# x509 device certificate with chain sample
+
+## How to run the sample
+
+1) Create IotHub and device to use in sample.
+2) Run the [AuthSetup.ps1](https://github.com/Azure-Samples/azure-iot-samples-csharp/blob/master/iot-hub/Samples/device/X509DeviceCertWithChainSample/AuthSetup.ps1) to setup the IoT hub and device with necessary certificates.
+
+```powershell
+.\AuthSetup.ps1 `
+-certFolderPath <Path to root cert> `
+-rootCertPassword <root certificate password> `
+-iotHubResourceGroup <IoT hub resource group> `
+-iotHubName <IoT hub name> `
+-deviceId <device name> 
+```
+
+## Important things to note
+
+1) Use the same IoT hub created in step 1 as input to step 2 and step 3.
+2) Use the same device created in step 1 as input to step 2 and step 3.
+3) Use the same password created in in step 2 as input to step 3.
```

Comment:
When I ran the script and provided my own password, it complained that:
AuthSetup.ps1: Cannot process argument transformation on parameter 'rootCertPassword'. Cannot convert the "password" value of type "System.String" to type "System.Security.SecureString".

I had to create a plain password then call Convert-ToSecureString then pass that to the script.

Reply:
Yes that is what is expected as it is a secure string input. But it would be good to call it out with an example in the readme.